main.py

```diff
@@ -93,27 +91,10 @@ def get_config():
     return config
 
 
-def init_env(config):
-    # set a debug environment variable CUBLAS_WORKSPACE_CONFIG to ":16:8" (may limit overall performance) or ":4096:8" (will increase library footprint in GPU memory by approximately 24MiB).
-    # https://docs.nvidia.com/cuda/cublas/index.html
-    os.environ['CUBLAS_WORKSPACE_CONFIG'] = ":4096:8"
-    if config.seed is not None:
-        if config.seed >= 0:
-            np.random.seed(config.seed)
-            torch.manual_seed(config.seed)
-            torch.set_deterministic(True)
-            torch.backends.cudnn.benchmark = False
-        else:
-            logging.warning(f'the random seed should be a non-negative integer')
-
-    config.device = None
-    if not config.cpu and torch.cuda.is_available():
-        config.device = torch.device('cuda')
-    else:
-        config.device = torch.device('cpu')
-        # https://github.com/pytorch/pytorch/issues/11201
-        torch.multiprocessing.set_sharing_strategy('file_system')
-    logging.info(f'Using device: {config.device}')
+def main():
+    config = get_config()
+    set_seed(seed=config.seed)
+    config.device = init_device(config.cpu)
```

Comment:
```suggestion
    config.device = init_device(use_cpu=config.cpu)
```
config.cpu sounds like we want to use cpu.. But "use_cpu" shows this means to use it or not